src/test/java/fi/hsl/transitdata/cancellation/util/BulletinUtilsTest.java

```diff
@@ -0,0 +1,117 @@
+                InternalMessages.Bulletin.Impact.CANCELLED,
+                InternalMessages.Bulletin.Priority.WARNING,
+                "HSL:1111", "HSL:2222");
+        
+        InternalMessages.Bulletin bulletinOther1 = createBulletin(
+                InternalMessages.Bulletin.Impact.IRREGULAR_DEPARTURES,
+                InternalMessages.Bulletin.Priority.WARNING,
+                "HSL:3333", "HSL:4444");
+        
+        InternalMessages.Bulletin bulletinMassCancellation2 = createBulletin(
+                InternalMessages.Bulletin.Impact.CANCELLED,
+                InternalMessages.Bulletin.Priority.WARNING,
+                "HSL:5555", "HSL:5555");
+        
+        InternalMessages.Bulletin bulletinOther2 = createBulletin(
+                InternalMessages.Bulletin.Impact.RETURNING_TO_NORMAL,
+                InternalMessages.Bulletin.Priority.INFO,
+                "HSL:6666", "HSL:7777");
+        
+        InternalMessages.Bulletin bulletinOther3 = createBulletin(
+                InternalMessages.Bulletin.Impact.DISRUPTION_ROUTE,
+                InternalMessages.Bulletin.Priority.SEVERE,
+                "HSL:8888", "HSL:9999");
+        
+        List<InternalMessages.Bulletin> inputBulletins = new ArrayList<>();
+        inputBulletins.add(bulletinMassCancellation1);
+        inputBulletins.add(bulletinOther1);
+        inputBulletins.add(bulletinMassCancellation2);
+        inputBulletins.add(bulletinOther2);
+        inputBulletins.add(bulletinOther3);
+        
+        List<InternalMessages.Bulletin> outputBulletins = BulletinUtils.filterMassCancellationsFromBulletins(inputBulletins);
+        
+        assertTrue(outputBulletins.size() == 2);
+    }
+    
+    @Test
+    public void parseCancellationDataFromBulletins() {
+        try (MockedStatic<TripUtils> tripUtils = Mockito.mockStatic(TripUtils.class)) {
+            
+            List<InternalMessages.TripInfo> trips = new ArrayList<>();
+            trips.add(TripUtilsTest.createTripInfo("HSL:4611", "HSL:4611_20240102_Ti_2_1415", "20240102", "1415", 1));
+            trips.add(TripUtilsTest.createTripInfo("HSL:4611", "HSL:4611_20240102_Ti_2_1515", "20240102", "1515", 1));
+            trips.add(TripUtilsTest.createTripInfo("HSL:1079", "HSL:1079_20240102_La_1_0734", "20240102", "0734", 2));
+            
+            tripUtils.when(() -> TripUtils.getTripInfos(
+                    any(List.class), any(LocalDateTime.class), any(LocalDateTime.class), anyString())).thenReturn(trips);
+            
+            InternalMessages.Bulletin bulletinMassCancellation = createBulletin(
+                    InternalMessages.Bulletin.Impact.CANCELLED,
+                    InternalMessages.Bulletin.Priority.WARNING,
+                    "HSL:4611", "HSL:1079");
+            
+            List<InternalMessages.Bulletin> inputBulletins = new ArrayList<>();
+            inputBulletins.add(bulletinMassCancellation);
```

Comment:
split this to own method `initializeTestTrips` since it's not really relevant to the test itself. Makes reading actual test easier